packages/sign_in_with_apple/lib/src/widgets/sign_in_with_apple_button.dart

```diff
@@ -20,79 +23,157 @@ class SignInWithAppleButton extends StatelessWidget {
         assert(iconAlignment != null),
         super(key: key);
 
+  /// The callback which will be called when the button is pressed.
   final VoidCallback onPressed;
 
+  /// The text which should be rendered inside the button.
+  ///
+  /// Defaults to `Sign in with Apple`.
   final String text;
 
+  /// The height of the button.
+  ///
+  /// This defaults to `44` according to the Apple Guidelines.
   final double height;
 
+  /// The style of the button according to Apple Guidelines.
+  ///
+  /// This defaults to [SignInWithAppleButtonStyle.black].
   final SignInWithAppleButtonStyle style;
 
+  /// The border radius of the button.
+  ///
+  /// Defaults to `8` pixels.
   final BorderRadius borderRadius;
 
+  /// How the icon should be aligned inside the button
+  ///
+  /// Defaults to [IconAlignment.center]
   final IconAlignment iconAlignment;
 
+  /// Returns the background color of the button based on the [style]
+  Color get _backgroundColor {
+    switch (style) {
+      case SignInWithAppleButtonStyle.black:
+        return Colors.black;
+      case SignInWithAppleButtonStyle.white:
+      case SignInWithAppleButtonStyle.whiteOutlined:
+        return Colors.white;
+    }
+
+    assert(false, 'Unknown style: $style');
+    return null;
+  }
+
+  /// Returns the contrast color to the [_backgroundColor]
+  ///
+  /// This is used for the text color and the icon color of the button
+  Color get _contrastColor {
+    switch (style) {
+      case SignInWithAppleButtonStyle.black:
+        return Colors.white;
+      case SignInWithAppleButtonStyle.white:
+      case SignInWithAppleButtonStyle.whiteOutlined:
+        return Colors.black;
+    }
+
+    assert(false, 'Unknown style: $style');
+    return null;
+  }
+
+  /// The decoration which should be applied to the inner container inside the button
+  ///
+  /// This allows to customize the border of the button
+  Decoration get _decoration {
+    switch (style) {
```

Comment:
The switch makes it nicer in case we decide to add more case to the enums in this file

Reply:
Absolutely 👍